File: Ruby/001_Program/main.rb
#!/usr/bin/env ruby

# Hello World - Basic Ruby Program
# Demonstrates basic Ruby syntax and output

puts "Hello, Ruby World!"
puts "=" * 40

# Variables
name = "Ruby Developer"
version = RUBY_VERSION

puts "Welcome to Ruby!"
puts "Hello, #{name}"
puts "Ruby Version: #{version}"

# Arrays
features = [
  "Object-Oriented Programming",
  "Dynamic Typing",
  "Elegant Syntax",
  "Rails Framework",
  "Gems and Libraries"
]

puts "\nRuby Features:"
features.each_with_index do |feature, index|
  puts "#{index + 1}. #{feature}"
end

# Hash
info = {
  language: "Ruby",
  creator: "Yukihiro Matsumoto",
  year: 1995,
  paradigm: "Object-Oriented"
}

puts "\nLanguage Information:"
info.each do |key, value|
  puts "#{key.to_s.capitalize}: #{value}"
end

# Simple method
def greet(name)
  "Hello, #{name}!"
end

puts "\n#{greet('World')}"

File: CPP/001_Program/main.cpp
#include <iostream>
#include <string>
using namespace std;

/**
 * Hello World - Basic C++ Program
 * Demonstrates basic C++ syntax, input/output, and variables
 */

int main() {
    // Basic output
    cout << "Hello, C++ World!" << endl;
    cout << "================================" << endl;

    // Variables
    string name;
    int age;

    // Input
    cout << "Enter your name: ";
    getline(cin, name);

    cout << "Enter your age: ";
    cin >> age;

    // Output
    cout << "\nWelcome, " << name << "!" << endl;
    cout << "You are " << age << " years old." << endl;

    // Display C++ features
    cout << "\nC++ Features:" << endl;
    cout << "- Object-Oriented Programming" << endl;
    cout << "- Templates and Generics" << endl;
    cout << "- Standard Template Library" << endl;
    cout << "- Memory Management" << endl;
    cout << "- High Performance" << endl;

    return 0;
}
